### ISSUE TITLE
Feature Request - Backup targets other than Google Drive.

### ISSUE THREAD

**Issue by 123dev** [sabeechen/hassio-google-drive-backup#354]
Would be really awesome to be able to backup to other targets.
More specifically 
- Azure Storage
- AWS S3
- Microsoft One Drive
...

Thanks

**Comment by Salvora** [comment]
+1 for OneDrive

**Comment by lbouriez** [comment]
+1 for OneDrive, google drive was good at the beginning but it's too expensive to keep many backup :/

**Comment by sabeechen** [comment]
My official policy on this is that I'd like to support more upload destinations, but at the moment I don't have the time to implement it.  Honestly I have trouble just keeping the current feature set polished in response to changes HA makes.  Then there is the problem of keeping the recognition I've already built up from calling it "Home Assistant Google Drive Backup".  Assuming I navigate all that, I'd like to consider making this work with:
* Dropbox
* OneDrive
* Azure
* AWS
* Samba
* FTP
* Nextcloud
* SFTP

**Comment by lbouriez** [comment]
Adding more would be amazing.
Google drive is so expensive compared to other solutions 🙄

**Comment by jant90** [comment]
While I'm perfectly happy with just Google Drive, if you really want to add more storage systems, maybe you could consider using [Rclone](https://rclone.org) for the backup process. It supports more cloud storage providers than you can think of and it would give you a single, unified way to backup to all those providers. It's very well maintained, actively tested by a large community, has no dependencies and is available for all architectures that HassOS runs on, so its single, pre-compiled binary can simply be added to your add-on's docker container and will run perfectly fine. You could issue commands through the CLI, or even use its [API over HTTP](https://rclone.org/rc/). This way you don't have to maintain any cloud provider API, not even the Google Drive one. Of course, all this would make more sense in a new add-on with a different name and branding.

**Comment by SirGoodenough** [comment]
Until sabeechen has more time, there is an RClone backup addon out there.
Check out this with an RClone option, an SCP option, and an RSync option.  I use it alongside this addon as my alternate for the 1-2-3 backup strategy.  
[ ikifar2012 /
remote-backup-addon ](https://github.com/ikifar2012/remote-backup-addon)

**Comment by jamesmyatt** [comment]
If this were refactored to use fsspec, then adding other backends would probably be easy. However that first part is going to be hard. https://filesystem-spec.readthedocs.io/en/latest/

**Comment by marcgarciamarti** [comment]
I would love to see OneDrive supported alongside with Google Backup!

**Comment by AlecRust** [comment]
Yeah this addon seems by far the best for managing backups, but it's a shame it's tied to only Google Drive.

As people's Home Assistant installation grows they often add more addons that are very large, and they want these backed up too. Google Drive lends itself better to smaller installations.

**Comment by caraar12345** [comment]
A bit of advice for Google Drive: give Google Workspace Business Standard a try, and use **shared drives**.

[Limits are file **count** based](https://support.google.com/a/users/answer/7338880?hl=en), not file size, so you can have up to 400,000 files, all of which are up to 750GB each.

So essentially, £10 a month for unlimited storage 🤓

**Comment by sabeechen** [comment]
Thats how I do it too.  Google seems to say there is a quota limit in my workspace dashboard, but they don't seem to do anything if I go over it, even _way_ over.  I suspect its a limit they'll enforce in the future only if too many people use it.

**Comment by EpicLPer** [comment]
Well, the Workspace limit hit me now and I'm moving stuff back over to DropBox again (after just half finishing doing the opposite...)

This AddOn supporting other locations would be awesome, I can't find any active AddOns right now that support DropBox sadly.

**Comment by rezeptpflichtig** [comment]
Uh, nextcloud support would be very nice 😗

**Comment by snickers2k** [comment]
+1
maybe just use rclone as the upload tool in the middle? it's already supporting all these solutions, inclusive local/network targets

**Comment by snickers2k** [comment]
> Well, the Workspace limit hit me now and I'm moving stuff back over to DropBox again (after just half finishing doing the opposite...)
> 
> This AddOn supporting other locations would be awesome, I can't find any active AddOns right now that support DropBox sadly.

sorry to ask here, but since i'm also running out of space and do not have more than 2 backups of hassio, i'm thinking of moving back to google workspaces again (after having lost many TB on GSuite on unlimted drive). Did you really used the SHARED Drive? Since this should not have a limit on sizes , but only on 400k file-count ?!
Can anyone confirm?

**Comment by svoop** [comment]
Still very much a topic.

**Comment by svoop** [comment]
Nothing changed, not stale at all. And please add "generic S3" to the list, that's AWS S3 with configurable endpoint URL which brings home tons of alternative S3 compatible cloud storage providers.

**Comment by m4v3r1ckNl** [comment]
> My official policy on this is that I'd like to support more upload destinations, but at the moment I don't have the time to implement it. Honestly I have trouble just keeping the current feature set polished in response to changes HA makes. Then there is the problem of keeping the recognition I've already built up from calling it "Home Assistant Google Drive Backup". Assuming I navigate all that, I'd like to consider making this work with:
> 
>     * Dropbox
> 
>     * OneDrive
> 
>     * Azure
> 
>     * AWS
> 
>     * Samba
> 
>     * FTP
> 
>     * Nextcloud
> 
>     * SFTP

I'm very happy with the current state-of-the-art "Home Assistant Google Drive Backup" as is. This is my way-to-go to back up my Home Assistant to Home Assistant instance and Google Drive. Nevertheless, offering users more options for different destination is always a good thing for a high UX.

Thank you very much for maintaining a high quality - flawless operations since my initial installation when starting with Home Assistant - backup package for Home Assistant. Never bothered to switch - trial-and-error - to the (NC) new core variant within Home Assistant.

Keep up the good work and kudos for your dedication to your package.

Cheers!

**Comment by svoop** [comment]
To whom it may concern: The [S3-Compatible add-on via HACS](https://github.com/PhantomPhoton/S3-Compatible) supports a number of other S3 providers. It initially wouldn't work with Hetzner Object Storage, but after some updates on both ends, my backups to this EU based service work flawlessly.
